src/chat/chatter.py

```diff
@@ -265,127 +265,79 @@ def doubleClicked(self, item):
+            action.triggered.connect(action_connect)  # Triggers
+            menu.addAction(action)
 
         # only for us. Either way, it will display our avatar, not anyone avatar.
-        if self.lobby.client.login == self.name :
-            menu.addAction(actionSelectAvatar)
-            menu.addSeparator()
+        if self.lobby.client.login == self.name:
+            menu_add("Select Avatar", self.selectAvatar)
 
         # power menu
         if self.lobby.client.power > 1:
             # admin and mod menus
-            actionAddAvatar = QtWidgets.QAction("Assign avatar", menu)
-            menu.addAction(actionAddAvatar)
-            actionAddAvatar.triggered.connect(self.addAvatar)
+            menu_add("Assign avatar", self.addAvatar, True)
 
             if self.lobby.client.power == 2:
-                action_inspect_in_mordor = QtWidgets.QAction("Send the Orcs", menu)
-                menu.addAction(action_inspect_in_mordor)
 
                 def send_the_orcs():
                     route = Settings.get('mordor/host')
-
                     if self.id != -1:
                         QtGui.QDesktopServices.openUrl(QUrl("{}/users/{}".format(route, self.id)))
                     else:
                         QtGui.QDesktopServices.openUrl(QUrl("{}/users/{}".format(route, self.name)))
 
-                action_inspect_in_mordor.triggered.connect(send_the_orcs)
+                menu_add("Send the Orcs", send_the_orcs, True)
+                menu_add("Close Game", lambda: self.lobby.client.closeFA(self.name))
+                menu_add("Close FAF Client", lambda: self.lobby.client.closeLobby(self.name))
+
+        # Aliases link
+        menu_add("View Aliases", self.viewAliases, True)
+
+        # Joining hosted or live Game
+        if self.lobby.client.login != self.name:  # Don't allow self to be invited to a game, or join one
+            if self.name in client.instance.urls:
 
-                actionCloseFA = QtWidgets.QAction("Close Game", menu)
-                menu.addAction(actionCloseFA)
-                actionCloseFA.triggered.connect(lambda: self.lobby.client.closeFA(self.name))
+                url = client.instance.urls[self.name]
+                if url.scheme() == "fafgame":
+                    menu_add("Join hosted Game", self.joinInGame, True)
+                elif url.scheme() == "faflive":
+                    menu_add("View live Replay", self.viewReplay, True)
 
-                actionCloseLobby = QtWidgets.QAction("Close FAF Client", menu)
-                menu.addAction(actionCloseLobby)
-                actionCloseLobby.triggered.connect(lambda: self.lobby.client.closeLobby(self.name))
+        # Replays in vault
+        if self.id != -1:  # not for irc user
+            menu_add("View Replays in Vault", self.viewVaultReplay, True)
 
-            menu.addSeparator()
+        # Friends and Foes Lists
+        def f_action_connect(f, irc_f):  # Irc or not Irc, that's the Question
```

Comment:
I don't know if f_action_connect is descriptive enough, how about player_or_irc_action?